Sources/SpeziFHIR/FHIRResource/FHIRResource+Copy.swift

```diff
@@ -0,0 +1,28 @@
+//
+// This source file is part of the Stanford Spezi open source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import ModelsDSTU2
+import ModelsR4
+
+
+extension FHIRResource {
+    /// Creates an (expensive) copy of the underlying FHIR reference types by encoding and decoding the FHIR resource in JSON.
+    ///
+    /// - Warning: The computational complexity of creating a copy should only be used in when creating a new copy is strictly nescessary.
+    public func copy() throws -> Self {
```

Comment:
Maybe we can achieve the same thing with less resources by using the Mirror API?

Reply:
Sure, might be an option in the future.